services/records_to_agol.py

```diff
@@ -29,6 +30,32 @@ def format_filter_date(date_from_args):
     return "1970-01-01" if not date_from_args else arrow.get(date_from_args).isoformat()
 
 
+def resilient_layer_request(func, args, max_retries=MAX_RETRIES):
+    """
+    An ArcGIS request wrapper to enable re-trying. The wrapper will only suppress timeout
+    exceptions from the Rest API. Our separate response handler utility catches API
+    errors.
+    """
+    attempts = 0
+    while True:
+        attempts += 1
+        try:
+            return func(**args)
+        except Exception as e:
+            """
+            The ArcGIS Python API raises a generic Exception class on timeout, so we parse the
+            Exception content to be sure we pass the right case. The timeout message string
+            we want to ignore:
+                `Exception: Your request has timed out.`
+            """
+            if "timed out" not in e.__str__().lower() or attempts == max_retries:
```

Comment:
i think this logic is sound. to be safe i could make the second conditional `attempts >= max_retries` ?